core/modules/image/image.install

```diff
@@ -80,18 +80,101 @@ function image_requirements($phase) {
-      if (function_exists('imagefilter') && function_exists('imagerotate')) {
-        $requirements['image_gd']['severity'] = REQUIREMENT_OK;
+        if ($supported) {
+          $supported_formats[] = $format;
+          $descriptions[] = array(
+            'data' => t('@format image format supported.', array('@format' => $format)),
+            'class' => array('status-ok'),
+          );
+        }
+        else {
+          $unsupported_formats[] = $format;
+          $descriptions[] = array(
+            'data' => t('@format image format not supported.', array('@format' => $format)),
+            'class' => array('status-error'),
+          );
+        }
       }
-      else {
+
+      if ($unsupported_formats) {
         $requirements['image_gd']['severity'] = REQUIREMENT_WARNING;
-        $requirements['image_gd']['description'] = t('The GD Library for PHP is enabled, but was compiled without support for functions used by the rotate and desaturate effects. It was probably compiled using the official GD libraries from http://www.libgd.org instead of the GD library bundled with PHP. You should recompile PHP --with-gd using the bundled GD library. See <a href="http://www.php.net/manual/book.image.php">the PHP manual</a>.');
+        $fix_info[] = t('If you want to add support for any unsupported image format, check the <a href="@url">PHP GD installation documentation</a>.', array('@url' => 'https://www.php.net/manual/image.installation.php'));
+      }
+
+      // Check for support of image effects.
+      $check_effects = array(
+        'Desaturate' => 'imagefilter',
+        'Rotate' => 'imagerotate',
+      );
+      $unsupported_effects = array();
+      foreach ($check_effects as $effect => $function) {
+        if (!function_exists($function)) {
+          $requirements['image_gd']['severity'] = REQUIREMENT_WARNING;
+          $unsupported_effects[] = $effect;
+          $descriptions[] = array(
+            'data' => t('@effect effect not supported.', array('@effect' => $effect)),
+            'class' => array('status-error'),
+          );
+        }
+        else {
+          $descriptions[] = array(
+            'data' => t('@effect effect supported.', array('@effect' => $effect)),
+            'class' => array('status-ok'),
+          );
+        }
+      }
+      if (!empty($unsupported_effects)) {
+        $fix_info[] = t('The GD library was compiled without support for functions required by certain effects. It was probably compiled using the official GD libraries from http://www.libgd.org instead of the GD library bundled with PHP. If you want to add support for these effects, recompile PHP --with-gd using the bundled GD library. See <a href="http://www.php.net/manual/book.image.php">the PHP manual</a>.');
       }
+
+      $formats_and_effects = theme('item_list', array('items' => $descriptions, 'title' => t('Supported image formats and effects'), 'attributes' => array('class' => array('requirements'))));
```

Comment:
Under the heading "Supported image formats and effects" may be listed supported and unsupported things. I'd suggest to replace "Supported" with "Support of" in the heading: `Support of image formats and effects`

Reply:
Good point @olafgrabienski 👍🏼 ...not sure if either of these options are ideal. I have an idea to move this text from being the title of the list of image format/effects to being appended to the value of the requirement as a status (whether all or only some effects/formats are supported).

I'll reply to the issue, and add some screenshots as well.